### ISSUE TITLE
Automatically add keys for icon path array

### ISSUE THREAD

**Issue by tomdohnal** [chakra-ui/chakra-ui#2007]
# Feature request

## Is your feature request related to a problem? Please describe.

When I create an icon using the `createIcon(...)` function and pass an array of paths as the `path` parameter, I have to manually specify the `key` prop for each of the paths.
```javascript

export const GoogleIcon = createIcon({
    displayName: 'GoogleIcon',
    viewBox: '0 0 533.5 544.3',
    path: [
        <path
            key={0}
            d="M533.5 278.4c0-18.5-1.5-37.1-4.7-55.3H272.1v104.8h147c-6.1 33.8-25.7 63.7-54.4 82.7v68h87.7c51.5-47.4 81.1-117.4 81.1-200.2z"
            fill="#4285f4"
        />,
        <path
            key={1}
            d="M272.1 544.3c73.4 0 135.3-24.1 180.4-65.7l-87.7-68c-24.4 16.6-55.9 26-92.6 26-71 0-131.2-47.9-152.8-112.3H28.9v70.1c46.2 91.9 140.3 149.9 243.2 149.9z"
            fill="#34a853"
        />,
        <path
            key={2}
            d="M119.3 324.3c-11.4-33.8-11.4-70.4 0-104.2V150H28.9c-38.6 76.9-38.6 167.5 0 244.4l90.4-70.1z"
            fill="#fbbc04"
        />,
        <path
            key={3}
            d="M272.1 107.7c38.8-.6 76.3 14 104.4 40.8l77.7-77.7C405 24.6 339.7-.8 272.1 0 169.2 0 75.1 58 28.9 150l90.4 70.1c21.5-64.5 81.8-112.4 152.8-112.4z"
            fill="#ea4335"
        />,
    ],
})
```

## Describe the solution you'd like
I think it'd be cool if Chakra did that for me. I believe it could be done quite easily just wrapping the individual path elements using `React.Fragment` and passing the `key` property to it.
I'm happy to work on it myself if it's something you'd want to incorporate.
## Describe alternatives you've considered
The alternative is to stick with the current state - it's not such a big of a deal to pass the `key` prop manually.

**Comment by hazem3500** [comment]
You can just wrap all the paths in a fragment as so
```tsx
export const GoogleIcon = createIcon({
    displayName: 'GoogleIcon',
    viewBox: '0 0 533.5 544.3',
    path: (
        <>
            <path
                d="M533.5 278.4c0-18.5-1.5-37.1-4.7-55.3H272.1v104.8h147c-6.1 33.8-25.7 63.7-54.4 82.7v68h87.7c51.5-47.4 81.1-117.4 81.1-200.2z"
                fill="#4285f4"
            />
            <path
                d="M272.1 544.3c73.4 0 135.3-24.1 180.4-65.7l-87.7-68c-24.4 16.6-55.9 26-92.6 26-71 0-131.2-47.9-152.8-112.3H28.9v70.1c46.2 91.9 140.3 149.9 243.2 149.9z"
                fill="#34a853"
            />
            <path
                d="M119.3 324.3c-11.4-33.8-11.4-70.4 0-104.2V150H28.9c-38.6 76.9-38.6 167.5 0 244.4l90.4-70.1z"
                fill="#fbbc04"
            />
            <path
                d="M272.1 107.7c38.8-.6 76.3 14 104.4 40.8l77.7-77.7C405 24.6 339.7-.8 272.1 0 169.2 0 75.1 58 28.9 150l90.4 70.1c21.5-64.5 81.8-112.4 152.8-112.4z"
                fill="#ea4335"
            />
        </>
    ),
});
```
would this solve your problem?

**Comment by ljosberinn** [comment]
I'd recommend that approach too, I'm [using it myself](https://github.com/ljosberinn/personal-react-boilerplate/blob/master/src/client/routes/Index/icons.tsx#L32) aswell. 

The [implementation](https://github.com/chakra-ui/chakra-ui/blob/develop/packages/icon/src/create-icon.tsx#L42) of `createIcon` is fairly trivial, I wouldn't be opposed to introducing `React.Children.map` but if `key`s are the only reason, I don't see it warranted at this point.

**Comment by Philzen** [comment]
@tomdohnal just wanted to share my finding that this was eventually implemented for #5129

es-lint will complain though, so in my case i put an `// eslint-disable-line` at the end of each path and now enjoy two saved lines of code and less indentation compared to the fragment workaround :rocket:
